titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/DefaultV3JobOperations.java

```diff
@@ -399,6 +401,69 @@ public Completable recordTaskPlacement(String taskId, Function<Task, Task> chang
                 .orElse(Observable.error(JobManagerException.taskNotFound(taskId)));
     }
 
+    @Override
+    public Observable<Void> moveServiceTask(String sourceJobId, String targetJobId, String taskId) {
+        if (!jobManagerConfiguration.isMoveTaskApiEnabled()) {
+            throw JobManagerException.notEnabled("Move task");
+        }
+        return Observable.defer(() -> {
+            Pair<ReconciliationEngine<JobManagerReconcilerEvent>, EntityHolder> fromEngineTaskPair =
+                    reconciliationFramework.findEngineByChildId(taskId).orElseThrow(() -> JobManagerException.taskNotFound(taskId));
+
+            ReconciliationEngine<JobManagerReconcilerEvent> engineFrom = fromEngineTaskPair.getLeft();
+            Job<ServiceJobExt> jobFrom = engineFrom.getReferenceView().getEntity();
+
+            // Validate the source job is a service job
+            if (!JobFunctions.isServiceJob(jobFrom)) {
+                throw JobManagerException.notServiceJob(jobFrom.getId());
+            }
+
+            // Validate that the task belongs to source job id
+            if (!jobFrom.getId().equals(sourceJobId)) {
+                throw JobManagerException.taskJobMismatch(taskId, sourceJobId);
+            }
+
+            // Validate that we are moving across different jobs
+            if (jobFrom.getId().equals(targetJobId)) {
+                throw JobManagerException.sameJobs(jobFrom.getId());
+            }
+
+            // TODO(fabio): validate that source and destination jobs are compatible (same container specs)
```

Comment:
Is this going to be in another PR?

Reply:
yes, please read the description of the PR :-)